src/main/java/com/google/cloud/tools/maven/RunMojo.java

```diff
@@ -201,11 +211,56 @@
 
   @Override
   public void execute() throws MojoExecutionException, MojoFailureException {
+    handleAppYamlsDeprecation();
     verifyAppEngineStandardApp();
     getAppEngineFactory().devServerRunSync().run(this);
   }
 
   /**
+   * <p>If &lt;appYamls&gt; is explicitly set by the user, we'll display a warning.</p>
+   * <p>If &lt;appYamls&gt; is explicitly set by the user and &lt;services&gt; is not (i.e. not
+   * present in the configuration; <i>note: the </i>{@code services}<i> field will contain the
+   * default value, so we cannot simply check if it's empty</i>), then we'll assign the value of
+   * &lt;appYamls&gt; to &lt;services&gt; to be used in the run configuration.</p>
+   * <p>If both &lt;appYamls&gt; and &lt;services&gt; are explicitly set in the configuration, we'll
+   * throw an error.</p>
+   *
+   * @throws MojoExecutionException if both &lt;appYamls&gt; and &lt;services&gt; are explicitly set
+   *     in the configuration
+   */
+  private void handleAppYamlsDeprecation() throws MojoExecutionException {
+    if (!appYamls.isEmpty()) {
+      // no default value, so it was set by the user explicitly
+      getLog().warn("<appYamls> is deprecated, use <services> instead.");
+      if (servicesUsesTheDefaultValue()) {
+        getLog().warn("The value of <appYamls> will override the default value of <services>");
+        services = appYamls;
+      } else {
+        throw new MojoExecutionException("Both <appYamls> and <services> are defined."
+            + " <appYamls> is deprecated, use <services> only.");
+      }
+    }
+  }
+
+  private boolean servicesUsesTheDefaultValue() throws MojoExecutionException {
+    MavenProject project = (MavenProject)getPluginContext().get("project");
+    Plugin ourPlugin = project.getPlugin("com.google.cloud.tools:appengine-maven-plugin");
+    Object configuration = ourPlugin.getConfiguration();
+    if (configuration instanceof org.apache.maven.shared.utils.xml.Xpp3Dom) {
```

Comment:
this seems very convoluted just to figure out if this setting was provided by the user.